docs/client.rst

```diff
@@ -106,6 +106,25 @@ is not encoded by library. Note that ``+`` is not encoded::
                            params='key=value+1') as r:
             assert str(r.url) == 'http://httpbin.org/get?key=value+1'
 
+.. note::
+
+   *aiohttp* internally performs URL canonization before sending request.
+
+   Canonization encodes *host* part by :term:`IDNA` codec and applies
+   :term:`requoting` to *path* and *query* parts.
+
+   For example ``URL('http://example.com/путь%30?a=%31')`` is converted to
+   ``URL('http://example.com/%D0%BF%D1%83%D1%82%D1%8C/0?a=1')``.
+
+   Sometimes canonization is not desirable if server accepts exact
+   representation and doesn't requote URL itself.
+
+   To disable canonization use ``encoded=True`` parameter for URL construction::
+
+      await session.get(URL('http://example.com/%30', encoded=True))
+
+   Passing *params* overrides ``encoded=True``, never use both options.
```

Comment:
On second thought, may be put that line in `.. warning` block? Would be hard to pass by.

Reply:
Put warning next after note section?

Reply:
Yes.